core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -55,7 +55,7 @@ class BrokerLifecycleManager(
   val config: KafkaConfig,
   val time: Time,
   val threadNamePrefix: Option[String],
-  val isZkBroker: Boolean = false
+  val zkBrokerEpochSupplier: Option[() => Long] = None
```

Comment:
this might be a little too clever I think. maybe we can just have a `boolean isZkBroker` and then a `() => Long` zkBrokerEpochSupplier ? 

I would really like to distinguish between the "not zk broker" and "zk broker doesn't have a broker epoch for some reason" cases